udata/tests/dataset/test_dataset_model.py

```diff
@@ -235,3 +235,62 @@ def test_url_is_stripped(self):
         url = 'http://www.somewhere.com/with/spaces/   '
         dataset = DatasetFactory(resources=[ResourceFactory(url=url)])
         self.assertEqual(dataset.resources[0].url, url.strip())
+
+
+class LicenseModelTest(DBTestMixin, TestCase):
+    def setUp(self):
+        super(LicenseModelTest, self).setUp()
+        # Feed the DB with random data to ensure true matching
+        LicenseFactory.create_batch(3)
+
+    def test_not_found(self):
+        found = License.guess('should not be found')
+        self.assertIsNone(found)
+
+    def test_exact_match_by_id(self):
+        license = LicenseFactory()
+        found = License.guess(license.id)
+        self.assertIsInstance(found, License)
+        self.assertEqual(license.id, found.id)
+
+    def test_exact_match_by_id_with_spaces(self):
+        license = LicenseFactory()
+        found = License.guess(' {0} '.format(license.id))
+        self.assertIsInstance(found, License)
+        self.assertEqual(license.id, found.id)
+
+    def test_exact_match_by_url(self):
+        license = LicenseFactory()
+        found = License.guess(license.url)
+        self.assertIsInstance(found, License)
+        self.assertEqual(license.id, found.id)
+
+    def test_exact_match_by_title(self):
+        license = LicenseFactory()
+        found = License.guess(license.title)
+        self.assertIsInstance(found, License)
+        self.assertEqual(license.id, found.id)
+
+    def test_exact_match_by_title_with_spaces(self):
+        license = LicenseFactory()
+        found = License.guess(' {0} '.format(license.title))
+        self.assertIsInstance(found, License)
+        self.assertEqual(license.id, found.id)
+
+    def test_match_by_title_with_low_edit_distance(self):
+        license = LicenseFactory(title='License')
+        found = License.guess('Licence')
+        self.assertIsInstance(found, License)
+        self.assertEqual(license.id, found.id)
+
+    def test_match_by_title_with_extra_inner_space(self):
+        license = LicenseFactory(title='xx yy')
+        found = License.guess('xx  yy')  # 2 spaces instead of 1
+        self.assertIsInstance(found, License)
+        self.assertEqual(license.id, found.id)
+
+    def test_match_by_title_with_mismatching_case(self):
+        license = LicenseFactory(title='x' * 6)
+        found = License.guess('xXxXxX')
+        self.assertIsInstance(found, License)
+        self.assertEqual(license.id, found.id)
```

Comment:
What does it change? Test are written to isolate a single case the string is chosen to specificaly highlight this case (in this one, that case changes does not impact the match)